### PR TITLE
Small fixes and optimizations

### LINKED ISSUE #256: Any possibility to make backend uia faster
i have a very simple script that click few buttons in calculator app. If I use backend win32 executing script takes about 16 sec, but if I use backend uia same script takes 1 min 35 sec. Below examples scripts. Each script is executed 4 times.

script below uses buttons text to find it

```
app = Application()
appconnect = app.connect(title='Calculator')
app_dlg = appconnect.top_window_()
app_dlg.window_(best_match='3').Click()
```

script below uses buttons automatinID

```
app = Application(backend='uia')
appconnect = app.connect(title='Calculator')
app_dlg = appconnect.top_window_()
app_dlg.child_window(auto_id='button5').click_input()
```

### PR DESCRIPTION
**Pull request by vasily-v-ryabov**
* Fix crash #269 
* Partially fix #256 with small optimizations
* Cosmetic changes

Had no chance to write more tests yet.

### CODE REVIEW

File: pywinauto/base_wrapper.py [hunk 1]
```diff
@@ -797,11 +797,13 @@ def type_keys(
         .. _keyboard: pywinauto.keyboard.html
         """
         self.verify_actionable()
+        friendly_class_name = self.friendly_class_name()
 
         if pause is None:
             pause = Timings.after_sendkeys_key_wait
 
         if set_foreground:
+            # self.top_level_parent().set_focus()
```
Comment (airelil) [inline pywinauto/base_wrapper.py]: Debug line ? Should be removed

File: pywinauto/findwindows.py [hunk 2]
```diff
@@ -200,10 +201,11 @@ def find_elements(class_name=None,
         # if not given a parent look for all children of the desktop
         if not parent:
             parent = backend_obj.element_info_class()
+        elif isinstance(parent, BaseWrapper):
+            parent = parent.element_info
```
Comment (airelil) [inline pywinauto/findwindows.py]: I'm not sure I understand it. Is the check for the instance of ```backend_obj.generic_wrapper_class``` at line 180 not enough ?

File: pywinauto/unittests/test_application.py [hunk 3]
```diff
@@ -434,12 +434,12 @@ def _cabinetwclass_exist():
 
     def test_wait_cpu_usage_lower_uia(self):
         """Test that wait_cpu_usage_lower() works correctly for UIA"""
-        app = Application(backend='uia')
-        app.start('notepad.exe')
+        self.app = Application(backend='uia')
+        self.app.start('notepad.exe')
```
Comment (airelil) [inline pywinauto/unittests/test_application.py]: A minor note. Why ```app``` as a local variable was not enough ?

**Comment by airelil** [comment]
Oh. I just noticed a number of faliures on Appveyor. ```test_wait_cpu_usage_lower_uia``` should be run conditionally only when UIA backend is enabled

File: pywinauto/unittests/test_application.py (hunk 3 above)
Comment (vasily-v-ryabov) [inline pywinauto/unittests/test_application.py]: It adds a Codacy warning. But it's minor. Will try to revert back and see if this check is red or not.

**Comment by airelil** [comment]
Maybe we need to use mock more extensievly If we want to increase the coverage. For example:
```
        while self.cpu_usage(usage_interval) > threshold:
            if time.time() - start_time > timeout:
                raise RuntimeError('Waiting CPU load <= {}% timed out!'.format(threshold))
```
can be easily tested with mocking ```self.cpu_usage``` method

**Pull request merged**